### PR TITLE
feat: add export for whole query result

### PR DESCRIPTION
**Pull request by invisal**
I have added the mock-up design

You follow these code for export
https://github.com/invisal/libsql-studio/blob/develop/gui/src/components/query-result-table.tsx#L370

### CODE REVIEW

File: gui/src/components/export/export-result-button.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import OptimizeTableState from "../table-optimized/OptimizeTableState";
+import { useCallback } from "react";
+
+export default function ExportResultButton({
+  data,
+}: {
+  data: OptimizeTableState;
+}) {
+  const onExportClicked = useCallback(() => {
+    // do something here
```
Comment (invisal) [inline gui/src/components/export/export-result-button.tsx:19]: Please add implementation here

Comment (thormengkheang) [reply gui/src/components/export/export-result-button.tsx:19]: thank i'll work on it when i got the time

File: gui/src/lib/export-helper.ts [hunk 2]
```diff
@@ -44,3 +44,38 @@ export function exportRowsToExcel(records: unknown[][]) {
 
   return result.join("\r\n");
 }
+
+export function exportRowsToJson(
```
Comment (avevotsira) [inline gui/src/lib/export-helper.ts:48]: lacking table name here too

Comment (invisal) [reply gui/src/lib/export-helper.ts:48]: We don't need table name for JSON

File: gui/src/components/table-optimized/OptimizeTableState.tsx [hunk 3]
```diff
@@ -335,6 +335,10 @@ export default class OptimizeTableState {
     return !!this.data[index]?.isRemoved;
   }
 
+  getAllRows() {
+    return this.data;
+  }
```
[the remark this answers is not in the document]
Comment (invisal) [reply gui/src/components/table-optimized/OptimizeTableState.tsx:340]: It is okay.

**Review by invisal — COMMENTED** [review]
CSV does not work properly. Try to open export.csv in Excel

**Comment by invisal** [comment]
LGTM

**Pull request merged**